Mainnet/AddressMapper/AddressMapper.Tests/AddressMapperTests.cs

```diff
@@ -0,0 +1,145 @@
+
+            mContractState.Setup(x => x.Message).Returns(new Message(contractAddress, primaryAddress, 0));
+
+            var contract = new AddressMapper(mContractState.Object, ownerAddress);
+
+            Assert.Throws<SmartContractAssertException>(() => contract.MapAddress(secondaryAddress));
+        }
+
+        [Fact]
+        public void MappAddress_Maps_PrimaryAddres()
+        {
+            persistentState.SetStruct($"MappingInfo:{secondaryAddress}", new MappingInfo { Primary = primaryAddress, Status = (int)Status.NoStatus });
+
+            mContractState.Setup(x => x.Message).Returns(new Message(contractAddress, primaryAddress, 0));
+
+            var contract = new AddressMapper(mContractState.Object, ownerAddress);
+
+            contract.MapAddress(secondaryAddress);
+
+            Assert.Equal(new MappingInfo { Primary = primaryAddress, Status = (int)Status.Pending }, persistentState.GetStruct<MappingInfo>($"MappingInfo:{secondaryAddress}"));
+        }
+
+        [Fact]
+        public void Approve_Fails_If_Called_By_None_Admin()
+        {
+            mContractState.Setup(x => x.Message).Returns(new Message(contractAddress, primaryAddress, 0));
+
+            var contract = new AddressMapper(mContractState.Object, ownerAddress);
+
+            Assert.Throws<SmartContractAssertException>(() => contract.Approve(secondaryAddress));
+        }
+
+        [Theory]
+        [InlineData(Status.Approved)]
+        [InlineData(Status.NoStatus)]
+        public void Approve_Fails_If_State_Is_Not_InPending_State(Status status)
+        {
+            persistentState.SetStruct($"MappingInfo:{secondaryAddress}", new MappingInfo { Primary = primaryAddress, Status = (int)status });
+
+            mContractState.Setup(x => x.Message).Returns(new Message(contractAddress, ownerAddress, 0));
+
+            var contract = new AddressMapper(mContractState.Object, ownerAddress);
+
+            Assert.Throws<SmartContractAssertException>(() => contract.Approve(secondaryAddress));
+        }
+
+        [Fact]
+        public void Approve_Approves_Mapping()
+        {
+            persistentState.SetStruct($"MappingInfo:{secondaryAddress}", new MappingInfo { Primary = primaryAddress, Status = (int)Status.Pending });
+
+            mContractState.Setup(x => x.Message).Returns(new Message(contractAddress, ownerAddress, 0));
+
+            var contract = new AddressMapper(mContractState.Object, ownerAddress);
+
+            contract.Approve(secondaryAddress);
+
+            Assert.Equal(secondaryAddress, persistentState.GetAddress($"SecondaryAddress:{primaryAddress}"));
+
+            Assert.Equal(new MappingInfo { Primary = primaryAddress, Status = (int)Status.Approved }, persistentState.GetStruct<MappingInfo>($"MappingInfo:{secondaryAddress}"));
+
+            mContractLogger.Verify(m => m.Log(mContractState.Object, new AddressMappedLog { Primary = primaryAddress, Secondary = secondaryAddress }));
```

Comment:
Same concept here, probably better to assert and verify that persistent state was actually set rather than the checking these values equal each other.

Reply:
`persistentState` is not a mocked object. It is `InMemoryState`.